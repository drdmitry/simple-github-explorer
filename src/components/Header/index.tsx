import type { FC } from "react";

import SearchBar from "~/components/SearchBar";
import useStore from "~/store/useStore";

export const Header: FC = () => {
  const { touched } = useStore();

  return (
    <header className="w-full p-4">
      {touched === false && (
        <div className="container flex flex-col m-auto pt-0 md:pt-14 lg:pt-28 items-center mb-4 sm:mb-8 md:mb-20 transition ease-in-out duration-200">
          <h1 className="text-3xl font-extrabold tracking-tight text-gray-500 max-[370px]:font-bold max-[370px]:text-[1.2rem] sm:text-[3rem] md:text-[4rem] lg:text-[5rem]">
            Simple <span className="text-sky-500">GitHub</span> Explorer
          </h1>
        </div>
      )}
      <div className="max-w-3xl m-auto flex flex-col sm:flex-row items-center gap-4">
        {touched && (
          <h1 className="text-3xl max-[370px]:text-[1.2rem] sm:text-xs text-gray-500 font-bold sm:font-bold flex flex-row sm:flex-col gap-2 sm:gap-0 items-center">
            <span>Simple</span>
            <span className="text-sky-500">GitHub</span>
            <span>Explorer</span>
          </h1>
        )}
        <SearchBar />
      </div>
    </header>
  );
};

export default Header;
