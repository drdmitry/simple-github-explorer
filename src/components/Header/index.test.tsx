import React from "react";
import { render } from "@testing-library/react";

import Header from "./";

describe("Header", () => {
  test("renders Header", () => {
    // Render the Header component
    const { container } = render(<Header />);
    const headerElement = container.querySelector("header");
    expect(headerElement).toBeInTheDocument();
  });

  test("renders Header unchanged", () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
});
