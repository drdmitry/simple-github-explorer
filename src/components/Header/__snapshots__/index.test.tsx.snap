// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders Header unchanged 1`] = `
<div>
  <header
    class="w-full p-4"
  >
    <div
      class="container flex flex-col m-auto pt-0 md:pt-14 lg:pt-28 items-center mb-4 sm:mb-8 md:mb-20 transition ease-in-out duration-200"
    >
      <h1
        class="text-3xl font-extrabold tracking-tight text-gray-500 max-[370px]:font-bold max-[370px]:text-[1.2rem] sm:text-[3rem] md:text-[4rem] lg:text-[5rem]"
      >
        Simple 
        <span
          class="text-sky-500"
        >
          GitHub
        </span>
         Explorer
      </h1>
    </div>
    <div
      class="max-w-3xl m-auto flex flex-col sm:flex-row items-center gap-4"
    >
      <div
        class="w-full flex flex-col sm:flex-row gap-1"
      >
        <div
          class="relative w-full"
        >
          <div
            class="absolute h-10 w-10 px-2 flex items-center"
          >
            <svg
              class="text-gray-600"
              fill="none"
              height="24"
              stroke="currentColor"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1"
              />
            </svg>
          </div>
          <input
            autocapitalize="off"
            autocomplete="off"
            autocorrect="off"
            class="bg-gray-100 rounded-md border border-gray-300 py-2 px-4 pl-10 block w-full outline-none focus:shadow hover:border-gray-400 focus:border:gray-400"
            placeholder="Search username"
            spellcheck="false"
            type="text"
            value=""
          />
        </div>
        <button
          class="bg-sky-500 bg-gradient-to-br from-sky-300 to-sky-500 hover:to-sky-600 active:to-sky-700 text-white shadow-sm font-bold py-2 px-4 rounded-md focus:outline-none focus:shadow-outline disabled:from-sky-100 disabled:to-sky-200 disabled:cursor-no-drop"
          disabled=""
          type="button"
        >
          Search
        </button>
      </div>
    </div>
  </header>
</div>
`;
